src/context/WalletContext.tsx

```diff
@@ -0,0 +1,36 @@
+import { createContext } from 'react';
+
+export enum WalletTypes {
+  METAMASK = 'METAMASK',
+  NONE = 'NONE',
+}
+
+export enum SignatureTypes {
+  EIP712 = 'EIP712',
+  EVM = 'EVM',
+}
+
+export type WalletConnectionOpts = {
+  chainId: string;
+  walletType: WalletTypes;
+};
+
+export type SignOpts = {
+  chainId: string;
+  signatureType: SignatureTypes;
+  payload: unknown; // todo: msg? transaction payload? this structure would be different based on the signatureType
+};
+
+export type WalletContextType = {
+  walletAddress: string;
+  walletChainId: string;
```

Comment:
Maybe eventually this becomes a white-listed enum of supported chain id's, but this works too.